client/src/pages/result/Result.js

```diff
@@ -48,6 +48,7 @@ export default function Result() {
 			method: "POST",
 			headers: {
 				"Content-Type": "application/json",
+				"Request-Source": 1
```

Comment:
Could you please explain / link some docs about the approach?

I think the idea was to use the request's origin and have a list of allowed hosts in the backend which we validate.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin

The Origin header can not be modified programmatically. But you could put "Request-Source" header in postman and make the request successfully which is not what we want.

Reply:
That was a solution that just came to mind. Your solution makes a lot more sense.
I've implemented it.